tests/unit/scripts/test_lint.py

```diff
@@ -158,111 +161,91 @@
         platform = $(some-script foo)
     [[baz]]
         platform = `no backticks`
-"""
-
-LINT_TEST_FILE += (
+""" + (
     '\nscript = the quick brown fox jumps over the lazy dog '
-    'until it becomes clear that this line is far longer the 79 characters.')
+    'until it becomes clear that this line is far longer the 79 characters.'
+)
 
 
-@pytest.fixture()
-def create_testable_file(monkeypatch, capsys):
-    def _inner(test_file, checks, ignores=None):
-        if ignores is None:
-            ignores = []
-        monkeypatch.setattr(Path, 'read_text', lambda _: test_file)
-        checks = parse_checks(checks, ignores)
-        check_cylc_file(Path('x'), Path('x'), checks)
-        return capsys.readouterr(), checks
-    return _inner
+def lint_text(text, checks, ignores=None, modify=False):
+    checks = parse_checks(checks, ignores)
+    counter = {}
+    messages = []
+    outlines = [
+        line
+        for line in lint(
+            'flow.cylc',
+            iter(text.splitlines()),
+            checks,
+            counter,
+            modify=modify,
+            write=messages.append
+        )
+    ]
```

Comment:
```suggestion
    outlines = list(
        lint(
            'flow.cylc',
            iter(text.splitlines()),
            checks,
            counter,
            modify=modify,
            write=messages.append
        )
    )
```

Reply:
(Discussed in person, happy enough not to use `list()` in the test - it may have issues with some generators, not sure)